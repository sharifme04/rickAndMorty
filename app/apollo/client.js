import {ApolloClient, InMemoryCache} from '@apollo/client';
import fetch from 'cross-fetch';
import {createHttpLink} from 'apollo-link-http';

// Initialize Apollo Client
const client = new ApolloClient({
  link: createHttpLink({
    uri: 'https://rickandmortyapi.com/graphql/',
    fetch: fetch,
  }),
  cache: new InMemoryCache(),
});
export default client;
