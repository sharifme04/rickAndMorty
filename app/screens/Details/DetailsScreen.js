import React from 'react';
import {
  ScrollView,
  ActivityIndicator,
  Image,
  View,
  SafeAreaView,
  Text,
} from 'react-native';
import {theme} from '../../theme/theme';
import styles from './Style';

const DetailsScreen = ({route}) => {
  const {id, image, gender, species, episode} = route?.params || '';

  return (
    <SafeAreaView style={styles.flexContiainer}>
      <View style={styles.flexContiainer}>
        <ScrollView>
          <View style={styles.body}>
            <View>
              {image ? (
                <Image
                  style={styles.image}
                  source={{
                    uri: `${image}`,
                  }}
                />
              ) : (
                <ActivityIndicator
                  size="large"
                  color={theme.colors.headerBackground}
                />
              )}
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionDescription}>Gender: {gender}</Text>
              <Text style={styles.sectionDescription}>Species: {species}</Text>
              <View>
                <Text style={styles.sectionDescription}>Episode: </Text>
                {episode?.map((e, i) => (
                  <View key={id + i} style={styles.episodeContainer}>
                    <Text style={styles.sectionDescription}>
                      Name: {e.name}
                    </Text>
                    <Text style={styles.sectionDescription}>
                      Date: {e.air_date}
                    </Text>
                  </View>
                ))}
              </View>
            </View>
          </View>
        </ScrollView>
      </View>
    </SafeAreaView>
  );
};

export default DetailsScreen;
