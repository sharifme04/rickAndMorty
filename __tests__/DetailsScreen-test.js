import 'react-native';
import React from 'react';
import {MockedProvider} from '@apollo/client/testing';
import DetailsScreen from '../app/screens/Details/DetailsScreen';
import renderer from 'react-test-renderer';

const details = {
  gender: 'Male',
  id: '1',
  image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
  species: 'Human',
  episode: [
    {air_date: 'December 2, 2013', name: 'Pilot', __typename: 'Episode'},
    {
      air_date: 'December 9, 2013',
      name: 'Lawnmower Dog',
      __typename: 'Episode',
    },
  ],
};

test('DetailsScreen', () => {
  const tree = renderer
    .create(
      <MockedProvider addTypename={false}>
        <DetailsScreen />
      </MockedProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('DetailsScreen with details data', () => {
  const tree = renderer
    .create(
      <MockedProvider addTypename={false}>
        <DetailsScreen details={details} />
      </MockedProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('DetailsScreen with details empty object ', () => {
  const tree = renderer
    .create(
      <MockedProvider addTypename={false}>
        <DetailsScreen details={{}} />
      </MockedProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
