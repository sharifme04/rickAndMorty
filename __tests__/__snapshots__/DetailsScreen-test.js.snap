// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 40,
              "marginTop": 25,
            }
          }
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#f4511e"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginTop": 60,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginVertical": 1,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Gender: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginVertical": 1,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Species: 
            </Text>
            <View>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginVertical": 1,
                    "padding": 5,
                    "textAlign": "center",
                  }
                }
              >
                Episode: 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`DetailsScreen with details data 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 40,
              "marginTop": 25,
            }
          }
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#f4511e"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginTop": 60,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginVertical": 1,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Gender: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginVertical": 1,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Species: 
            </Text>
            <View>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginVertical": 1,
                    "padding": 5,
                    "textAlign": "center",
                  }
                }
              >
                Episode: 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`DetailsScreen with details empty object  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 40,
              "marginTop": 25,
            }
          }
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#f4511e"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginTop": 60,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginVertical": 1,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Gender: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginVertical": 1,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Species: 
            </Text>
            <View>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginVertical": 1,
                    "padding": 5,
                    "textAlign": "center",
                  }
                }
              >
                Episode: 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
